src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -389,6 +389,12 @@ public void InstallService(string serviceName, string serviceDisplayName, string
                     throw new InvalidOperationException(StringUtil.Loc("OperationFailed", nameof(CreateService), GetLastError()));
                 }
 
+                //invoke the service with special argument, that tells it to register an event log trace source (need to run as an admin)
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    processInvoker.ExecuteAsync(string.Empty, agentServiceExecutable, $"init \"{serviceDisplayName}\"", null, default(System.Threading.CancellationToken)).GetAwaiter().GetResult();                    
```

Comment:
you don't have to pass servicedisplayname, the service.exe control what name to use.